### PR TITLE
fix: Desktop should show update prompt for 1.0.1 release

### PR DESCRIPTION
**Pull request by gsusmi**
An update available prompt needs to be displayed when next version is made available for desktop.

Ref: ICU-1560.

### CODE REVIEW

File: ui/desktop/electron-app/src/helpers/app-updater.js [hunk 1]
```diff
@@ -33,12 +33,12 @@ const findLatestVersion = (url) => {
 
 // Find zip archive for update
 const findUpdateArchive = (version) => {
-  const url = `${releasesUrl}boundary-desktop_${version}/boundary-desktop_${version}_darwin_amd64.zip`;
+  const url = `${releasesUrl}${version}/boundary-desktop_${version}_darwin_amd64.zip`;
```
Comment (calcaide) [inline ui/desktop/electron-app/src/helpers/app-updater.js:36]: This comment is out of topic, but at some point we might need to make the architecture `_darwin_amd64` dynamic. I am thinking on windows client o M1 apple chips (arm64)

Comment (gsusmi) [reply ui/desktop/electron-app/src/helpers/app-updater.js:36]: Agree. Incoming windows support PR configures release url based on os platform and architecture.

Comment (calcaide) [reply ui/desktop/electron-app/src/helpers/app-updater.js:36]: Awesome!

**Review by calcaide — APPROVED** [review]
LGTM

**Review by randallmorey — APPROVED** [review]

**Pull request merged**